src/coordinator/storage/local/storage.go

```diff
@@ -22,32 +22,39 @@ package local
 
 import (
 	"context"
+	goerrors "errors"
+	"fmt"
+	"sync"
 	"time"
 
 	"github.com/m3db/m3db/src/coordinator/block"
 	"github.com/m3db/m3db/src/coordinator/errors"
 	"github.com/m3db/m3db/src/coordinator/models"
 	"github.com/m3db/m3db/src/coordinator/storage"
 	"github.com/m3db/m3db/src/coordinator/util/execution"
-	"github.com/m3db/m3db/src/dbnode/client"
+	"github.com/m3db/m3db/src/dbnode/storage/index"
+	xerrors "github.com/m3db/m3x/errors"
 	"github.com/m3db/m3x/ident"
 	"github.com/m3db/m3x/pool"
 	xtime "github.com/m3db/m3x/time"
 )
 
+const (
+	initRawFetchAllocSize = 32
+)
+
+var (
+	errNoLocalClustersFulfillsQuery = goerrors.New("no clusters can fulfill query")
+)
+
 type localStorage struct {
-	session    client.Session
-	namespace  ident.ID
+	clusters   Clusters
 	workerPool pool.ObjectPool
 }
 
 // NewStorage creates a new local Storage instance.
-func NewStorage(session client.Session, namespace string, workerPool pool.ObjectPool) storage.Storage {
-	return &localStorage{
-		session:    session,
-		namespace:  ident.StringID(namespace),
-		workerPool: workerPool,
-	}
+func NewStorage(clusters Clusters, workerPool pool.ObjectPool) storage.Storage {
```

Comment:
I think this is the same as in #785 right? Can you rebase this on master now that it's merged?